app/static/site.js

```diff
@@ -166,18 +166,18 @@ function refresh_img(imgUrl) {
 
   // update img.src
   document.querySelectorAll(`[src="${imgUrl}"]`).forEach(function (e) {
-    e.src = imgUrl;
+    e.src = imgUrl.replace("img/", "snapshot/");
```

Comment:
This is going to cause flickering. You're going to need to fetch and pre decode the new url before swapping it in.

Reply:
Would `tmp.src = url.replace("img/", "snapshot/");`  on 192 fix that?